File: cash.c
#include<cs50.h>
#include<math.h>
#include<stdio.h>

int main(void)
{
    float dollers;
    int i=0;   //Declare variable for counting the number of coins needed
    do
    {
        dollers = get_float("Change owed: ");  // Get non negetive change value
    }
    while(dollers < 0);
    int coins = round(dollers * 100);    // Round the float to integer
    while(coins > 0)
    {
       if(coins >= 25)     //check while greater than 25
       {
           coins -= 25;
           i++;
       }
       else if(coins >= 10)    //check whether greater than 10
       {
           coins -= 10;
           i++;
       }
       else if(coins >= 5)    //check whether grearher than 5
       {
           coins -= 5;
           i++;
       }
       else    //check for the rest 1
       {
           coins -= 1;
           i++;
       }
               
    }
    printf("%d\n", i);    //print the number of coins needed
}
© 2020 GitHub, Inc.

File: pset6/cash/cash.py
from cs50 import get_float

while True:
    owed = get_float("Change owed: ")
    if owed > 0:
        break
coins = owed * 100
count = 0
if coins >= 25:
    count += (coins - (coins % 25)) / 25
    coins = coins % 25
if coins >= 10:
    count += (coins - (coins % 10)) / 10
    coins = coins % 10
if coins >= 5:
    count += (coins - (coins % 5)) / 5
    coins = coins % 5
count += coins
print(int(count))